webapps/ui/cockpit/client/scripts/components/variables/variable-add-dialog.js

```diff
@@ -103,28 +101,34 @@ var Controller = [
 
       var data = angular.extend({}, newVariable);
 
-      if (operation.variables) {
-        operation.variables = operation.variables.filter(variable => {
-          return data.name !== variable.variable.name;
-        });
+      const variables = resolved.variables;
+      if (variables && ($scope.isBatchOperation || $scope.isMigration)) {
+        for (let i = 0; i < variables.length; i++) {
+          if (data.name === variables[i].variable.name) {
+            variables.splice(i, 1);
+          }
+        }
 
-        operation.variables.push({
+        variables.push({
           variable: data,
           valid: true,
-          changed: false
+          changed: false,
+          showFailures: true
         });
 
-        $scope.$root.$broadcast('variable.added');
+        $scope.$root.$broadcast('variable.added', data.name);
```

Comment:
(Not in the scope of this ticket)
Using the root scope can lead to problems when we have multiple variable tables on one page. I don't think we have that anywhere at the moment, so we can just keep using it